cyipopt/scipy_interface.py

```diff
@@ -127,30 +159,49 @@ def gradient(self, x, **kwargs):
 
     def constraints(self, x):
         con_values = []
-        for fun, args in zip(self._constraint_funs, self._constraint_args):
-            con_values.append(fun(x, *args))
+        for i, (fun, args) in enumerate(
+                zip(self._constraint_funs, self._constraint_args)):
+            if self._constraint_funs_with_jacs[i]:
+                con_values.append(
+                    self.__evaluate_con_fun_with_jac(i, fun, x, *args)[0])
+            else:
+                con_values.append(fun(x, *args))
         return np.hstack(con_values)
 
     def jacobian(self, x):
         con_values = []
-        for fun, args in zip(self._constraint_jacs, self._constraint_args):
-            con_values.append(fun(x, *args))
+        for i, (fun, jac, args) in enumerate(
+                zip(self._constraint_funs, self._constraint_jacs, self._constraint_args)):
+            if self._constraint_funs_with_jacs[i]:
+                con_values.append(
+                    self.__evaluate_con_fun_with_jac(i, fun, x, *args)[1])
+            else:
+                con_values.append(jac(x, *args))
         return np.vstack(con_values)
 
-    def intermediate(
-            self,
-            alg_mod,
-            iter_count,
-            obj_value,
-            inf_pr,
-            inf_du,
-            mu,
-            d_norm,
-            regularization_size,
-            alpha_du,
-            alpha_pr,
-            ls_trials
-            ):
+    def __evaluate_con_fun_with_jac(self, i, con_fun, x, *args):
+        if self.last_x is None or not np.all(self.last_x == x):
```

Comment:
Should this be a `np.isclose` check for `x` rather than a direct check for equality?

Reply:
It's really intended to be a check for equality since `np.isclose(self.last_x, x)` would return `True` if the new point `x` lies in a small neighborhood  of `self.last_x`. However, we must ensure that it's the same point. It's more or less identical to the `evaluate_fun_with_grad()` method.